openghg/retrieve/_search.py

```diff
@@ -605,11 +605,29 @@ def _base_search(**kwargs: Any) -> SearchResults:
             logger.exception(msg=error_msg)
             raise ValueError(error_msg)
 
-        # Here we create a dictionary of the metadata keyed by the Datasource UUID
-        # we'll create a pandas DataFrame out of this in the SearchResult object
-        # for better printing / searching within a notebook
+        # Due to versions of OpenGHG < 0.6.0 not storing the object store
+        # in the metadata we need to pass that information in here for backwards
+        # compatibility.
         metadata = {r["uuid"]: r for r in metastore_records}
 
+        updated_metadata = {}
+        store_data = {"object_store": bucket}
```

Comment:
How is bucket being defined (and how do we want it to be defined) - at the moment it looks like it's just the last item in the loop on L589?

Reply:
Yeah we iterate over the dictionary `readable_buckets` that's a store_name: store_path pair.

Reply:
But won't `bucket` just be set to the last value taken from `readable_buckets` from the loop? Is that what is expected?

Reply:
Ok - right this part is in the loop - I missed the nesting here.